src/main/java/com/artipie/rpm/Rpm.java

```diff
@@ -81,25 +81,32 @@
      */
     private final NamingPolicy naming;
 
+    /**
+     * Hashing sum computation algorithm.
+     */
+    private final Digest dgst;
+
     /**
      * Ctor.
      * @param stg Storage
      */
     public Rpm(final Storage stg) {
-        this(stg, NamingPolicy.DEFAULT);
+        this(stg, NamingPolicy.DEFAULT, Digest.SHA256);
     }
 
     /**
      * Ctor.
      * @param stg The storage
      * @param naming RPM files naming policy
+     * @param dgst Hashing sum computation algorithm
      */
-    public Rpm(final Storage stg, final NamingPolicy naming) {
+    public Rpm(final Storage stg, final NamingPolicy naming, final Digest dgst) {
```

Comment:
@brastak is it possible to use different hash algs for name and for digest xml field?

Reply:
The code allows you to do this now, but it's should not be used. 
createrepo uses consistent hashing - one algorithm for everything. I can do the same, but I don't know what is the best (or at least acceptable) from eo/0cracy point of view:
- typical Spring scenario: algorithm injected in all reference places (really, code will stay the same, just auto-injection will guaranty consistency);
- builder pattern scenario: Rpm will be built by builder, Naming policy and Digest will be specified with different calls (withNamingPolicy / withDigest), consistency will be provided on the build() call;
- move responsibility scenario: move NamingPolicy.name() to the Repomd class. Then NamingPolicy will become enum by nature (the only purpose is to say use hashes or nor), Rpm/Repomd will do the logic based on digest and Naming Policy (can lead to code duplication);
- separate responsibility scenario: NamingPolicy becomes enum, new class (NamingProvider, for example) is created. NamingPolicy is passed to Rpm constructor, NamingProvider is created in the constructor based on Policy and Digest. Provider will generate names and will be used instead of NamingPolicy in the code. Looks like it's the best suitable for us, but it's overcomplicated from my point of view.

I mean, for me the first two options are intuitive, and the last two options are non-intuitive. But first two options are forbidden, aren't they? 
@g4s8 Do I need to change this with the last approach? Or you have some other suggestion?

Reply:
@brastak ok, let's keep it as is for now